execution/contexts/bond_context.go

```diff
@@ -0,0 +1,178 @@
+package contexts
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/hyperledger/burrow/acm"
+	"github.com/hyperledger/burrow/acm/acmstate"
+	"github.com/hyperledger/burrow/acm/validator"
+	"github.com/hyperledger/burrow/execution/exec"
+	"github.com/hyperledger/burrow/logging"
+	"github.com/hyperledger/burrow/permission"
+	"github.com/hyperledger/burrow/txs/payload"
+)
+
+type BondContext struct {
+	StateWriter  acmstate.ReaderWriter
+	ValidatorSet validator.ReaderWriter
+	Logger       *logging.Logger
+	tx           *payload.BondTx
+}
+
+// Execute a BondTx to add a new validator
+func (ctx *BondContext) Execute(txe *exec.TxExecution, p payload.Payload) error {
+	var ok bool
+	ctx.tx, ok = p.(*payload.BondTx)
+	if !ok {
+		return fmt.Errorf("payload must be BondTx, but is: %v", txe.Envelope.Tx.Payload)
+	}
+
+	// the account initiating the bond
+	account, err := ctx.StateWriter.GetAccount(ctx.tx.Input.Address)
+	if err != nil {
+		return err
+	}
+
+	// ensure pubKey of validator is set
+	val := ctx.tx.Validator
+	if err := GetIdentity(ctx.StateWriter, val); err != nil {
+		return fmt.Errorf("couldn't retrieve identity: %v", err)
+	}
+
+	// check if validator already exists
+	power, err := ctx.ValidatorSet.Power(*val.Address)
+	if err != nil {
+		return err
+	} else if power != nil && power.Cmp(big.NewInt(0)) == 1 && account.Address != *val.Address {
+		// we currently do not support delegated bonding
+		return fmt.Errorf("%s is already bonded", val.Address)
```

Comment:
Seems like the error message here should be more like the comment. Also what is the non-positive power check for. Why do we continue if we are bonding 0 but input address does not match validator address?

Reply:
I guess it does nothing, but seems odd to continue.